src/rpc/methods/eth.rs

```diff
@@ -876,7 +919,99 @@ fn recover_sig(sig: &Signature) -> Result<(EthBigInt, EthBigInt, EthBigInt)> {
         sig.bytes().get(64..65).expect("failed to get slice"),
     );
 
-    Ok((EthBigInt(r), EthBigInt(s), EthBigInt(v)))
+    Ok((r.into(), s.into(), v.into()))
+}
+
+fn recover_legacy_homestead_sig(sig: &Signature) -> Result<(EthBigInt, EthBigInt, EthBigInt)> {
```

Comment:
1. Let's avoid having hard logic in the `rpc` module. This should be used for schema cosmetics and re-using existing elements. I know it's not the case for now, but avoiding adding debt here will be helpful for a future refactor.
2. This looks pretty similar to https://github.com/ChainSafe/forest/blob/07b0a8791ca3da7a07dd129d15647c344943726f/src/eth/eip_155_transaction.rs#L43 - can we avoid logic duplication?

Reply:
Fixed.